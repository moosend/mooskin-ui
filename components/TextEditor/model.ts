import { IAllProps } from '@tinymce/tinymce-react';

export interface ITextEditorComponentProps extends Partial<IAllProps> {
	/** selects different element to attach the editor to */
	selector?: any;

	/** show/hide the menu bar */
	menubar?: boolean;

	/** personalization tags custom dropdown */
	personalizationTags?: {
		buttonLabel: string;
		id: string;
		tags: IPersonalizationTag[];
	};
}

export interface IPersonalizationTag {
	label: string;
	value: string;
}
