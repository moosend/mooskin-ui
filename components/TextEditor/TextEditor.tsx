import * as React from 'react';

import { Editor } from '@tinymce/tinymce-react';

// Models
import { Editor as TinyMCEEditor } from 'tinymce';
import { IPersonalizationTag, ITextEditorComponentProps } from './model';

/**
 * TextEditor
 */
export const TextEditor: React.FC<ITextEditorComponentProps> = (props) => {
	const getSetup = (editor: TinyMCEEditor) => {
		props.personalizationTags &&
			editor.ui.registry.addMenuButton(props.personalizationTags.id, {
				fetch: (callback) => {
					const items: any =
						props.personalizationTags &&
						props.personalizationTags.tags.map((item: IPersonalizationTag) => {
							return {
								onAction: () => {
									editor.insertContent(item.value);
								},
								text: item.label,
								type: 'menuitem',
							};
						});
					callback(items);
				},
				text: props.personalizationTags.buttonLabel,
			});
	};

	const getToolbar = () => {
		if (props.personalizationTags) {
			return `${props.toolbar} | ${props.personalizationTags.id}`;
		}
		return props.toolbar;
	};

	return (
		<Editor
			{...props}
			apiKey="f3vo81k6z7efry5af62a1l5nm882r1dyrqn2df1cugtsofwq"
			toolbar={getToolbar()}
			init={{
				setup: props.personalizationTags ? getSetup : undefined,
				...props.init,
				menubar: props.menubar,
				selector: props.selector,
			}}
		/>
	);
};

TextEditor.defaultProps = {
	// className: '',
	disabled: false,
	init: {},
	inline: false,
	menubar: false,
	plugins: [
		'lists advlist',
		'anchor',
		'autolink',
		'autoresize',
		'charmap',
		'code codesample',
		'directionality',
		'emoticons',
		'fullscreen',
		'help',
		'hr',
		'image imagetools',
		'insertdatetime',
		'link',
		'nonbreaking',
		'paste',
		'preview',
		'print',
		'quickbars',
		'searchreplace',
		'table',
		// 'template',
		'toc',
		'visualblocks',
		'wordcount',
	],
	// style: {},
	// toolbar: `bullist numlist | emoticons | code | image | anchor link | paste pastetext |
	// `
	toolbar: `newdocument |
            undo redo |
            cut copy paste pastetext |
            link unlink openlink anchor |
            numlist bullist |
            outdent indent |
            blockquote |
            alignleft alignright aligncenter alignjustify |
            bold italic strikethrough underline removeformat |
            formatselect |
            fontselect fontsizeselect forecolor backcolor |
            visualblocks |
            code |`,
};

TextEditor.displayName = 'TextEditor';
