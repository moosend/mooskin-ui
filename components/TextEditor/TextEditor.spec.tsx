import * as React from 'react';
import { TextEditor } from './TextEditor';

import { mount } from 'enzyme';

const fn = jest.fn();

describe.skip('TextEditor', () => {
	test('renders correctly', () => {
		const tree = mount(<TextEditor value="<p>TextEditor</p>" onEditorChange={fn} />);
		expect(tree).toMatchSnapshot();
	});

	test('renders inline correctly', () => {
		const tree = mount(<TextEditor inline value="<p>TextEditor</p>" onEditorChange={fn} />);
		expect(tree).toMatchSnapshot();
	});

	test('renders disabled correctly', () => {
		const tree = mount(<TextEditor disabled value="<p>TextEditor</p>" onEditorChange={fn} />);
		expect(tree).toMatchSnapshot();
	});

	test('renders with menubar correctly', () => {
		const tree = mount(<TextEditor menubar value="<p>TextEditor</p>" onEditorChange={fn} />);
		expect(tree).toMatchSnapshot();
	});
});
