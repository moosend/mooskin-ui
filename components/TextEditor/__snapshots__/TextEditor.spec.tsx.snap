// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextEditor renders correctly 1`] = `
<TextEditor
  disabled={false}
  init={Object {}}
  inline={false}
  menubar={false}
  onEditorChange={[MockFunction]}
  plugins={
    Array [
      "lists advlist",
      "anchor",
      "autolink",
      "autoresize",
      "charmap",
      "code codesample",
      "directionality",
      "emoticons",
      "fullscreen",
      "help",
      "hr",
      "image imagetools",
      "insertdatetime",
      "link",
      "nonbreaking",
      "paste",
      "preview",
      "print",
      "quickbars",
      "searchreplace",
      "table",
      "toc",
      "visualblocks",
      "wordcount",
    ]
  }
  toolbar="newdocument |
            undo redo |
            cut copy paste pastetext |
            link unlink openlink anchor |
            numlist bullist |
            outdent indent |
            blockquote |
            alignleft alignright aligncenter alignjustify |
            bold italic strikethrough underline removeformat |
            formatselect |
            fontselect fontsizeselect forecolor backcolor |
            visualblocks |
            code |"
  value="<p>TextEditor</p>"
>
  <Editor
    apiKey="f3vo81k6z7efry5af62a1l5nm882r1dyrqn2df1cugtsofwq"
    cloudChannel="5"
    disabled={false}
    init={
      Object {
        "menubar": false,
        "selector": undefined,
        "setup": undefined,
      }
    }
    inline={false}
    menubar={false}
    onEditorChange={[MockFunction]}
    plugins={
      Array [
        "lists advlist",
        "anchor",
        "autolink",
        "autoresize",
        "charmap",
        "code codesample",
        "directionality",
        "emoticons",
        "fullscreen",
        "help",
        "hr",
        "image imagetools",
        "insertdatetime",
        "link",
        "nonbreaking",
        "paste",
        "preview",
        "print",
        "quickbars",
        "searchreplace",
        "table",
        "toc",
        "visualblocks",
        "wordcount",
      ]
    }
    toolbar="newdocument |
            undo redo |
            cut copy paste pastetext |
            link unlink openlink anchor |
            numlist bullist |
            outdent indent |
            blockquote |
            alignleft alignright aligncenter alignjustify |
            bold italic strikethrough underline removeformat |
            formatselect |
            fontselect fontsizeselect forecolor backcolor |
            visualblocks |
            code |"
    value="<p>TextEditor</p>"
  >
    <textarea
      id="tiny-react_33179739021612904465037"
      style={
        Object {
          "visibility": "hidden",
        }
      }
    />
  </Editor>
</TextEditor>
`;

exports[`TextEditor renders disabled correctly 1`] = `
<TextEditor
  disabled={true}
  init={Object {}}
  inline={false}
  menubar={false}
  onEditorChange={[MockFunction]}
  plugins={
    Array [
      "lists advlist",
      "anchor",
      "autolink",
      "autoresize",
      "charmap",
      "code codesample",
      "directionality",
      "emoticons",
      "fullscreen",
      "help",
      "hr",
      "image imagetools",
      "insertdatetime",
      "link",
      "nonbreaking",
      "paste",
      "preview",
      "print",
      "quickbars",
      "searchreplace",
      "table",
      "toc",
      "visualblocks",
      "wordcount",
    ]
  }
  toolbar="newdocument |
            undo redo |
            cut copy paste pastetext |
            link unlink openlink anchor |
            numlist bullist |
            outdent indent |
            blockquote |
            alignleft alignright aligncenter alignjustify |
            bold italic strikethrough underline removeformat |
            formatselect |
            fontselect fontsizeselect forecolor backcolor |
            visualblocks |
            code |"
  value="<p>TextEditor</p>"
>
  <Editor
    apiKey="f3vo81k6z7efry5af62a1l5nm882r1dyrqn2df1cugtsofwq"
    cloudChannel="5"
    disabled={true}
    init={
      Object {
        "menubar": false,
        "selector": undefined,
        "setup": undefined,
      }
    }
    inline={false}
    menubar={false}
    onEditorChange={[MockFunction]}
    plugins={
      Array [
        "lists advlist",
        "anchor",
        "autolink",
        "autoresize",
        "charmap",
        "code codesample",
        "directionality",
        "emoticons",
        "fullscreen",
        "help",
        "hr",
        "image imagetools",
        "insertdatetime",
        "link",
        "nonbreaking",
        "paste",
        "preview",
        "print",
        "quickbars",
        "searchreplace",
        "table",
        "toc",
        "visualblocks",
        "wordcount",
      ]
    }
    toolbar="newdocument |
            undo redo |
            cut copy paste pastetext |
            link unlink openlink anchor |
            numlist bullist |
            outdent indent |
            blockquote |
            alignleft alignright aligncenter alignjustify |
            bold italic strikethrough underline removeformat |
            formatselect |
            fontselect fontsizeselect forecolor backcolor |
            visualblocks |
            code |"
    value="<p>TextEditor</p>"
  >
    <textarea
      id="tiny-react_19561344441612904465062"
      style={
        Object {
          "visibility": "hidden",
        }
      }
    />
  </Editor>
</TextEditor>
`;

exports[`TextEditor renders inline correctly 1`] = `
<TextEditor
  disabled={false}
  init={Object {}}
  inline={true}
  menubar={false}
  onEditorChange={[MockFunction]}
  plugins={
    Array [
      "lists advlist",
      "anchor",
      "autolink",
      "autoresize",
      "charmap",
      "code codesample",
      "directionality",
      "emoticons",
      "fullscreen",
      "help",
      "hr",
      "image imagetools",
      "insertdatetime",
      "link",
      "nonbreaking",
      "paste",
      "preview",
      "print",
      "quickbars",
      "searchreplace",
      "table",
      "toc",
      "visualblocks",
      "wordcount",
    ]
  }
  toolbar="newdocument |
            undo redo |
            cut copy paste pastetext |
            link unlink openlink anchor |
            numlist bullist |
            outdent indent |
            blockquote |
            alignleft alignright aligncenter alignjustify |
            bold italic strikethrough underline removeformat |
            formatselect |
            fontselect fontsizeselect forecolor backcolor |
            visualblocks |
            code |"
  value="<p>TextEditor</p>"
>
  <Editor
    apiKey="f3vo81k6z7efry5af62a1l5nm882r1dyrqn2df1cugtsofwq"
    cloudChannel="5"
    disabled={false}
    init={
      Object {
        "menubar": false,
        "selector": undefined,
        "setup": undefined,
      }
    }
    inline={true}
    menubar={false}
    onEditorChange={[MockFunction]}
    plugins={
      Array [
        "lists advlist",
        "anchor",
        "autolink",
        "autoresize",
        "charmap",
        "code codesample",
        "directionality",
        "emoticons",
        "fullscreen",
        "help",
        "hr",
        "image imagetools",
        "insertdatetime",
        "link",
        "nonbreaking",
        "paste",
        "preview",
        "print",
        "quickbars",
        "searchreplace",
        "table",
        "toc",
        "visualblocks",
        "wordcount",
      ]
    }
    toolbar="newdocument |
            undo redo |
            cut copy paste pastetext |
            link unlink openlink anchor |
            numlist bullist |
            outdent indent |
            blockquote |
            alignleft alignright aligncenter alignjustify |
            bold italic strikethrough underline removeformat |
            formatselect |
            fontselect fontsizeselect forecolor backcolor |
            visualblocks |
            code |"
    value="<p>TextEditor</p>"
  >
    <div
      id="tiny-react_90559280131612904465060"
    />
  </Editor>
</TextEditor>
`;

exports[`TextEditor renders with menubar correctly 1`] = `
<TextEditor
  disabled={false}
  init={Object {}}
  inline={false}
  menubar={true}
  onEditorChange={[MockFunction]}
  plugins={
    Array [
      "lists advlist",
      "anchor",
      "autolink",
      "autoresize",
      "charmap",
      "code codesample",
      "directionality",
      "emoticons",
      "fullscreen",
      "help",
      "hr",
      "image imagetools",
      "insertdatetime",
      "link",
      "nonbreaking",
      "paste",
      "preview",
      "print",
      "quickbars",
      "searchreplace",
      "table",
      "toc",
      "visualblocks",
      "wordcount",
    ]
  }
  toolbar="newdocument |
            undo redo |
            cut copy paste pastetext |
            link unlink openlink anchor |
            numlist bullist |
            outdent indent |
            blockquote |
            alignleft alignright aligncenter alignjustify |
            bold italic strikethrough underline removeformat |
            formatselect |
            fontselect fontsizeselect forecolor backcolor |
            visualblocks |
            code |"
  value="<p>TextEditor</p>"
>
  <Editor
    apiKey="f3vo81k6z7efry5af62a1l5nm882r1dyrqn2df1cugtsofwq"
    cloudChannel="5"
    disabled={false}
    init={
      Object {
        "menubar": true,
        "selector": undefined,
        "setup": undefined,
      }
    }
    inline={false}
    menubar={true}
    onEditorChange={[MockFunction]}
    plugins={
      Array [
        "lists advlist",
        "anchor",
        "autolink",
        "autoresize",
        "charmap",
        "code codesample",
        "directionality",
        "emoticons",
        "fullscreen",
        "help",
        "hr",
        "image imagetools",
        "insertdatetime",
        "link",
        "nonbreaking",
        "paste",
        "preview",
        "print",
        "quickbars",
        "searchreplace",
        "table",
        "toc",
        "visualblocks",
        "wordcount",
      ]
    }
    toolbar="newdocument |
            undo redo |
            cut copy paste pastetext |
            link unlink openlink anchor |
            numlist bullist |
            outdent indent |
            blockquote |
            alignleft alignright aligncenter alignjustify |
            bold italic strikethrough underline removeformat |
            formatselect |
            fontselect fontsizeselect forecolor backcolor |
            visualblocks |
            code |"
    value="<p>TextEditor</p>"
  >
    <textarea
      id="tiny-react_97260822951612904465065"
      style={
        Object {
          "visibility": "hidden",
        }
      }
    />
  </Editor>
</TextEditor>
`;
